src/ray/core_worker/test/direct_task_transport_test.cc

```diff
@@ -183,6 +183,25 @@ class MockActorCreator : public ActorCreatorInterface {
   ~MockActorCreator() {}
 };
 
+class MockLeasePolicy : public LeasePolicyInterface {
```

Comment:
Can you also add a test to make sure that we respect the LeasePolicy on the first lease request (i.e., not on spillback)? No need to modify all the tests, but maybe just the tests that have spillback (TestSpillback and TestSpillbackRoundTrip)?

Reply:
@stephanie-wang I did add a couple of assertions on `lease_policy->num_lease_policy_consults` in `TestSpillback` and `TestSpillbackRoundTrip` that should cover that (see lines [792](https://github.com/ray-project/ray/blob/b54d291051749130251dec3cf16faed6c1ef8df9/src/ray/core_worker/test/direct_task_transport_test.cc#L792), [802](https://github.com/ray-project/ray/blob/b54d291051749130251dec3cf16faed6c1ef8df9/src/ray/core_worker/test/direct_task_transport_test.cc#L802-L803), [868](https://github.com/ray-project/ray/blob/b54d291051749130251dec3cf16faed6c1ef8df9/src/ray/core_worker/test/direct_task_transport_test.cc#L868-L869)), unless I'm missing something.